alma/views/js/alma-product-insurance.js

```diff
@@ -20,266 +20,275 @@
-                    if (event.eventType === 'updatedProductQuantity') {
-                        quantity = getQuantity();
-                        if (event.event) {
-                            quantity = event.event.target.value;
-                        }
-                        removeInsurance();
-                    }
-                    if (event.eventType === 'updatedProductCombination') {
-                        removeInsurance();
-                    }
-                    if (typeof event.selectedAlmaInsurance !== 'undefined' && event.selectedAlmaInsurance !== null) {
-                        insuranceSelected = true;
-                        addInputsInsurance(event);
-                    }
-                    if (typeof event.selectedInsuranceData !== 'undefined' && event.selectedInsuranceData) {
-                        removeInputInsurance();
-                    }
-                    if (addToCartFlow) {
-                        addToCart.click();
-                        insuranceSelected = false;
-                        addToCartFlow = false;
+                if (event.event !== undefined) {
+                    quantity = getQuantity();
+                }
+                if (event.eventType === 'updatedProductQuantity') {
+                    quantity = getQuantity();
+                    if (event.event) {
+                        quantity = event.event.target.value;
                     }
+                    removeInsurance();
                 }
-            );
-            prestashop.on(
-                'updatedProduct',
-                function () {
-                    document.querySelector('.qty [name="qty"]').value = quantity;
-                    productDetails = JSON.parse(document.querySelector('.alma-widget-insurance .js-product-details').dataset.product);
-                    productDetails.quantity_wanted = parseInt(quantity);
-                    document.querySelector('.alma-widget-insurance .js-product-details').dataset.product = JSON.stringify(productDetails);
-                    refreshWidget();
-                    addModalListenerToAddToCart();
+                if (event.eventType === 'updatedProductCombination') {
+                    removeInsurance();
                 }
-            );
-        }
-    });
-})(jQuery);
+                if (typeof event.selectedAlmaInsurance !== 'undefined' && event.selectedAlmaInsurance !== null) {
+                    insuranceSelected = true;
+                    addInputsInsurance(event);
+                }
+                if (typeof event.selectedInsuranceData !== 'undefined' && event.selectedInsuranceData) {
+                    removeInputInsurance();
+                }
+                if (addToCartFlow) {
+                    addToCart.click();
+                    insuranceSelected = false;
+                    addToCartFlow = false;
+                }
+            });
+            prestashop.on('updatedProduct', function (data) {
```

Comment:
I think you need also to set the input quantity here with the previous quantity after the selected insurance otherwise you lost the quantity and add 1 product for more insurance.